File: 01/01.hs
import System.IO  
import Control.Monad

main = do  
    contents <- readFile "input.txt"
    let theWords = lines contents
    let (finalPosition, finalNrZeroes) = foldl applyTurn (50,0) theWords
    print finalNrZeroes

getDirectedStep :: String -> Integer
getDirectedStep a = case a of
        (d:steps) -> case d of
                'R' -> 1 * read steps
                'L' -> -1 * read steps

-- Input is the current state (position, nrZeroes) and next turn as a string
applyTurn :: (Integer, Integer) -> String -> (Integer, Integer)
applyTurn (position, nrZeroes) nextTurn =
        let directedStep = getDirectedStep nextTurn
            newPosition = mod (position + directedStep) 100
            isZero = if newPosition == 0 then 1 else 0
        in (newPosition, nrZeroes+isZero)

File: 01/01.py


def zero_counter(filename):

    with open(filename) as file:
        zero_count = 0
        position = 50
        for line in file:
            direction = 1 if line.startswith("R") else -1
            steps = int(line[1:]) * direction
            position = (position + steps) % 100
            if position == 0:
                zero_count += 1
        print(zero_count)


if __name__ == "__main__":
    zero_counter("input.txt")